front/pages/api/data_sources/[user]/[name]/search.ts

```diff
@@ -0,0 +1,203 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { User, DataSource, Provider } from "@app/lib/models";
+import { getServerSession } from "next-auth/next";
+import { authOptions } from "@app/pages/api/auth/[...nextauth]";
+import { Op } from "sequelize";
+import { credentialsFromProviders } from "@app/lib/providers";
+import { parse_payload } from "@app/lib/http_utils";
+import { JSONSchemaType } from "ajv";
+import { Document } from "@app/lib/types";
+import { Result, Ok, Err } from "@app/lib/result";
+
+type TagsFilter = {
+  in?: string[];
+  not?: string[];
+};
+
+type TimestampFilter = {
+  gt?: number;
+  lt?: number;
+};
+
+type SearchFilter = {
+  tags?: TagsFilter;
+  timestamp?: TimestampFilter;
+};
+
+export type DatasourceSearchQuery = {
+  query: string;
+  top_k: number;
+  full_text: boolean;
+  timestamp_gt?: number;
+  timestamp_lt?: number;
+  tags_in?: string[];
+  tags_not?: string[];
+};
+
+const searchQuerySchema: JSONSchemaType<DatasourceSearchQuery> = {
+  type: "object",
+  properties: {
+    query: { type: "string" },
+    top_k: { type: "number" },
+    full_text: { type: "boolean" },
+    timestamp_gt: { type: "number", nullable: true },
+    timestamp_lt: { type: "number", nullable: true },
+    tags_in: { type: "array", items: { type: "string" }, nullable: true },
+    tags_not: { type: "array", items: { type: "string" }, nullable: true },
+  },
+  required: ["query", "top_k", "full_text"],
+};
+
+const { DUST_API } = process.env;
+
+type DatasourceSearchResponseBody = {
+  documents: Array<Document>;
+};
+
+export async function performSearch(
```

Comment:
Let's move the code inside the endpoint as having a function call is just more code for no use.

Reply:
Also that makes you return number as error type which is unatural and exactly what we'll work on when factoring this logic

Reply:
BTW since you're not implementing the UI yet this handler should not be in that PR should it?